pkg/datastore/postgres.go

```diff
@@ -92,75 +64,78 @@ const (
 )
 
 // NewPostgresStorage creates a new Storage which uses postgres.
-func NewPostgresStorage(logger *zap.Logger, host, port, user, password, dbname, sslmode string, ves ...VersionedJSONEntity) (*Datastore, error) {
+func NewPostgresDB(logger *zap.Logger, host, port, user, password, dbname, sslmode string, ves ...Entity) (*sqlx.DB, error) {
 	db, err := sqlx.Connect("postgres", fmt.Sprintf("host=%s port=%s user=%s dbname=%s password=%s sslmode=%s", host, port, user, dbname, password, sslmode))
 	if err != nil {
 		return nil, fmt.Errorf("unable to connect to database: %w", err)
 	}
-	types := make(map[string]VersionedJSONEntity)
 	for _, ve := range ves {
 		jsonField := ve.JSONField()
 		logger.Info("creating schema", zap.String("entity", jsonField))
-		_, err = db.Exec(ve.Schema())
+		_, err := db.Exec(ve.Schema())
 		if err != nil {
 			logger.Fatal("unable to create schema", zap.String("entity", jsonField), zap.Error(err))
 			return nil, err
 		}
-		types[jsonField] = ve
 	}
-	ds := &Datastore{
-		log:   logger,
-		db:    db,
-		sb:    squirrel.StatementBuilder.PlaceholderFormat(squirrel.Dollar).RunWith(db),
-		types: types,
+	return db, nil
+}
+
+// NewPostgresStorage creates a new Storage which uses postgres.
+func NewPostgresStorage[E Entity](logger *zap.Logger, db *sqlx.DB, e E) (Storage[E], error) {
```

Comment:
Could also be just called `New`, not so interesting for the callers if this is based on Postgres or something else?

Reply:
good point, done